File: js/Fe.js
//example functions
function on(){
	return "light on";
}

function off(){
	return "light off";
}

//it is used by fn class to find the function inside string. 
const jsonObject = {
	"function":["on","off"],
	"on":{
		"name":"on",
		"argument":[]
	},
	"off":{
		"name":"off",
		"argument":[]
	}
}

//used to execute function found by fn class.
function ex(funcName){
	if(funcName == "on"){
		return on();
		}
	if(funcName == "off"){
		return off();
		}
	else{
		return "error:2"
		}
}

//used to find the function inside string
var fn = {
  parse : function(text=0){
    try{
      if(text == 0){
        return "err:0"
      }
      else{
        var setA = text.toString().split(" ");
        var funcExtract = jsonObject["function"].filter(x => setA.includes(x));
        if(funcExtract.length <= 0){
          return "err:1";
        }
        else{
          return ex(jsonObject[funcExtract[0]]["name"]);
        }
      }
    }
    catch(err){
      return "err:0";
    }
  }
};

File: py/Fe.py
class functionExecutor:
    def functionCall(self,meth,argument=0):
        try:
            if argument == 0:
                return meth()
            else:
                return meth(argument)
        except Exception as e:
            return False
    def textParsing(self,text,jsonObject):
        try:
            splitText = text.split(" ")
            filterFunc = list(set(jsonObject["functions"]) & set(splitText))
            filterLen = len(filterFunc)
            if filterLen > 0:
                funcName = jsonObject[filterFunc[0]]["name"]
                funcArgs = jsonObject[filterFunc[0]]["argument"]
                argsLen = len(funcArgs)
                if argsLen <= 0:
                    return self.functionCall(funcName)
                if argsLen > 0:
                    argsVal = list(set(funcArgs) & set(splitText))
                    argsValLen = len(argsVal)
                    if argsValLen > 0:
                        return self.functionCall(funcName,argsVal[0])
                    if argsValLen <= 0:
                        return False
            if filterLen <= 0:
                return False
        except Exception as e:
            return False

